src/api.js

```diff
@@ -1,34 +1,80 @@
 import { API_BASE } from './config';
 
+import axios from 'axios';
+import { RemediationsApi, ResolutionsApi, RemediationsApiAxiosParamCreator } from '@redhat-cloud-services/remediations-client';
+
+/*
+ * TODO: replace these with generated clients
+ */
+import { doGet } from './Utilities/http';
 import urijs from 'urijs';
-import { doGet, doPost, doPatch, doDelete } from './Utilities/http';
 
 function url (...args) {
     const url = urijs(API_BASE).segment('remediations');
     args.forEach(segment => url.segment(segment));
     return url;
 }
 
-export const getRemediations = (sortBy = 'updated_at', sortDir = 'desc') => {
-    const sort = `${sortDir === 'desc' ? '-' : ''}${sortBy}`;
-    return doGet(url().query({ sort }));
-};
-
-export const getRemediation = id => doGet(url(id));
 export const getRemediationStatus = id => doGet(url(id, 'status'));
-export const createRemediation = data => doPost(url(), data);
-export const patchRemediation = (id, data) => doPatch(url(id), data).then(() => data);
-export const patchRemediationIssue = (id, issue, resolution) => doPatch(url(id, 'issues', issue), { resolution });
-export const deleteRemediation = id => doDelete(url(id));
-export const deleteRemediationIssue = (remediation, issue) => doDelete(url(remediation, 'issues', issue));
-export const getResolutions = issue => doGet(urijs(API_BASE).segment('resolutions').segment(issue));
+export function getHosts () {
+    return doGet('/api/inventory/v1/hosts');
+}
+/*
+ * end of TODO
+ */
 
-export function downloadPlaybook (id) {
-    const uri = url(id, 'playbook').toString();
-    window.open(uri);
+class HttpError extends Error {
+    constructor(description) {
+        super('Error communicating with the server');
+        this.description = description;
+    }
 }
 
-// this is here for demo purposes only
-export function getHosts () {
-    return doGet('/api/inventory/v1/hosts');
+async function authInterceptor (config) {
```

Comment:
This is really good, I am thinking if we could move this to FEC so overybody can use these functions in future.